KerrSchild analytic solution now works for different frames.

markscheel opened the pull request:
## Proposed changes

KerrSchild now can be used for multiple frames.

Its member functions are now instantiated for Frame::Grid and Frame::Inertial.

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->
<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

nilsvu commented on src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp:
```diff
@@ -246,34 +248,33 @@ class KerrSchild : public MarkAsAnalyticSolution {
-                                         double /*t*/,
-                                         tmpl::list<Tags...> /*meta*/) const
-      noexcept {
+  tuples::TaggedTuple<Tags...> variables(
```
I believe you only need to template this function on `Frame`, plus all the things that are currently templated on `DataType` in this class. Then the public interface can remain unchanged. Do you think that'll work?

markscheel replied:
Yes, this works!  The changes to KerrSchild are more complicated, but the only things that need to change are KerrSchild and the test.

markscheel committed:
KerrSchild: Allow evaluating in grid frame.

markscheel committed:
Test_KerrSchild: Add many tests for grid frame.

markscheel commented:
Made changes and rebased.  Since the changes were so extensive, I didn't bother with fixup commits and I just re-pushed the new version.

markscheel commented:
Fixed test that was failing on older versions of gcc (variable was set outside an 'if constexpr' but used only inside one branch of 'if constexpr') and squashed.

nilsvu commented:
@markscheel just like in #3005 I deleted the "do nothing" upgrade instructions because I'm preparing the release. Please add upgrade instructions if you think they would be useful here.
